apps/src/lib/ui/accounts/DeleteAccount.jsx

```diff
@@ -44,26 +44,29 @@ const DEFAULT_STATE = {
   deleteError: '',
 };
 
+const dependedUponForLogin = ({isTeacher, hasStudents, dependentStudents}) => {
+  return isTeacher && hasStudents && (dependentStudents.length > 0);
+};
+
 export default class DeleteAccount extends React.Component {
   static propTypes = {
     isPasswordRequired: PropTypes.bool.isRequired,
     isTeacher: PropTypes.bool.isRequired,
-    dependedUponForLogin: PropTypes.bool.isRequired,
+    dependentStudents: dependentStudentsShape,
     hasStudents: PropTypes.bool.isRequired,
   };
 
   constructor(props) {
     super(props);
-    const checkboxes = getCheckboxes(props.dependedUponForLogin, props.hasStudents);
+    const checkboxes = getCheckboxes(dependedUponForLogin(props), props.hasStudents);
     this.state = {...DEFAULT_STATE, checkboxes};
   }
 
   componentDidUpdate = (prevProps) => {
-    // If dependedUponForLogin or hasStudents props have changed,
-    // checkboxes need to update accordingly.
-    const {dependedUponForLogin, hasStudents} = this.props;
-    if (dependedUponForLogin !== prevProps.dependedUponForLogin || hasStudents !== prevProps.hasStudents) {
-      const checkboxes = getCheckboxes(dependedUponForLogin, hasStudents);
+    // If dependedUponForLogin has changed, checkboxes need to update accordingly.
```

Comment:
Noting here because this change looks a little weird: 

Before, `dependedUponForLogin` and `hasStudents` came in as props and were calculated separately, so that meant I needed to check if _either_ prop had changed and re-calculate checkboxes accordingly.

Now, `dependedUponForLogin` is a calculated property (function) that incorporates `isTeacher`, `dependentStudents` (which replaces `dependedUponForLogin` as a prop), and `hasStudents`, so we no longer need to do separate checks when re-calculating which checkboxes are relevant.